python/tests/unit/arcticdb/version_store/test_lazy_dataframe.py

```diff
@@ -0,0 +1,456 @@
+"""
+Copyright 2024 Man Group Operations Limited
+
+Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt.
+
+As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+"""
+import numpy as np
+import pandas as pd
+import pickle
+import pytest
+
+from arcticdb import col, LazyDataFrame, LazyDataFrameCollection, QueryBuilder, ReadRequest
+from arcticdb.util.test import assert_frame_equal
+
+
+def test_lazy_read(lmdb_library):
+    lib = lmdb_library
+    sym = "test_lazy_read"
+    df = pd.DataFrame(
+        {"col1": np.arange(10, dtype=np.int64), "col2": np.arange(100, 110, dtype=np.int64)}, index=pd.date_range("2000-01-01", periods=10)
+    )
+    lib.write(sym, df)
+    lib.write_pickle(sym, 1)
```

Comment:
I know this is just to increment the version number, but how does the LazyDataFrame behave when reading pickled data? Probably worth some docs

Reply:
Same as `QueryBuilder`, an exception is thrown. This is covered in the `QueryBuilder` docstring:
https://github.com/man-group/ArcticDB/blob/master/python/arcticdb/version_store/processing.py#L384